pkg/fqdn/name_manager_gc.go

```diff
@@ -133,6 +137,13 @@ func (n *NameManager) GC(ctx context.Context) error {
 	}
 	log.WithField(logfields.Controller, dnsGCJobName).Infof(
 		"FQDN garbage collector work deleted %d name entries: %s", namesCount, strings.Join(namesToCleanSlice, ","))
+
+	// Remove any now-stale ipcache metadata.
+	// Need to RLock here so we don't race on re-insertion.
+	n.RWMutex.RLock()
+	n.maybeRemoveMetadataLocked(maybeStaleIPs)
+	n.RWMutex.RUnlock()
```

Comment:
It seems both callers of `maybeRemoveMetadataLocked` look like that. Maybe move the read lock and the comment into `maybeRemoveMetadataLocked` so not to obstruct the main control flow with details about locking?

Reply:
Oh, good point.